kview/src/main/java/dev/ikm/komet/kview/klfields/KlFieldHelper.java

```diff
@@ -0,0 +1,147 @@
+        ReadOnlyKLFieldFactory rowf = ReadOnlyKLFieldFactory.getInstance();
+        List<ObservableField<?>> observableFields = new ArrayList<>();
+        Consumer<FieldRecord<Object>> updateUIConsumer = (fieldRecord) -> {
+
+            Node node = null;
+            int dataTypeNid = fieldRecord.dataType().nid();
+            ObservableField observableField = obtainObservableField(viewProperties, semanticEntityVersionLatest, fieldRecord);
+            observableFields.add(observableField);
+            if (dataTypeNid == TinkarTerm.COMPONENT_FIELD.nid()) {
+                // load a read-only component
+                KlComponentFieldFactory componentFieldFactory = new KlComponentFieldFactory();
+                node = componentFieldFactory.create(observableField, viewProperties.nodeView(), true).klWidget();
+            } else if (dataTypeNid == TinkarTerm.STRING_FIELD.nid() || fieldRecord.dataType().nid() == TinkarTerm.STRING.nid()) {
+                KlStringFieldFactory stringFieldTextFactory = new KlStringFieldFactory();
+                node = stringFieldTextFactory.create(observableField, viewProperties.nodeView(), true).klWidget();
+            } else if (dataTypeNid == TinkarTerm.COMPONENT_ID_SET_FIELD.nid()) {
+                node = rowf.createReadOnlyComponentSet(viewProperties, fieldRecord);
+            } else if (dataTypeNid == TinkarTerm.COMPONENT_ID_LIST_FIELD.nid()) {
+                node = rowf.createReadOnlyComponentList(viewProperties, fieldRecord);
+            } else if (dataTypeNid == TinkarTerm.DITREE_FIELD.nid()) {
+                node = rowf.createReadOnlyDiTree(viewProperties, fieldRecord);
+            } else if (dataTypeNid == TinkarTerm.FLOAT_FIELD.nid() || fieldRecord.dataType().nid() == TinkarTerm.FLOAT.nid()) {
+                KlFloatFieldFactory klFloatFieldFactory = new KlFloatFieldFactory();
+                node = klFloatFieldFactory.create(observableField, viewProperties.nodeView(), true).klWidget();
+            } else if (dataTypeNid == TinkarTerm.INTEGER_FIELD.nid() || fieldRecord.dataType().nid() == TinkarTerm.INTEGER_FIELD.nid()) {
+                KlIntegerFieldFactory klIntegerFieldFactory = new KlIntegerFieldFactory();
+                node = klIntegerFieldFactory.create(observableField, viewProperties.nodeView(), true).klWidget();
+            } else if (dataTypeNid == TinkarTerm.BOOLEAN_FIELD.nid()) {
+                KlBooleanFieldFactory klBooleanFieldFactory = new KlBooleanFieldFactory();
+                node = klBooleanFieldFactory.create(observableField, viewProperties.nodeView(), true).klWidget();
+            }
+//            else if (dataTypeNid == TinkarTerm.IMAGE_FIELD.nid()) {
+//                node = rowf.createReadOnlyComponentSet(viewProperties, fieldRecord);
+//                ObservableField<> observableField = obtainObservableField(viewProperties, semanticEntityVersionLatest, fieldRecord);
+//                observableFields.add(observableField);
+//            }
```

Comment:
This is comment block is really for notes when image field is going to be added.